webapp/src/main/resources/public/js/tests/RepositoryInputHelper.test.js

```diff
@@ -0,0 +1,112 @@
+      { assetExtension: 'js', integrityCheckerType: 'checker1' },
+      { assetExtension: 'css', integrityCheckerType: 'checker2' }
+    ]);
+  });
+  it('trims whitespace', () => {
+    expect(deserializeAssetIntegrityCheckers(' js : checker1 , css : checker2 ')).toEqual([
+      { assetExtension: 'js', integrityCheckerType: 'checker1' },
+      { assetExtension: 'css', integrityCheckerType: 'checker2' }
+    ]);
+  });
+});
+
+describe('serializeAssetIntegrityCheckers', () => {
+  it('returns empty string for non-array', () => {
+    expect(serializeAssetIntegrityCheckers(null)).toBe('');
+    expect(serializeAssetIntegrityCheckers(undefined)).toBe('');
+  });
+  it('serializes single pair', () => {
+    expect(serializeAssetIntegrityCheckers([
+      { assetExtension: 'js', integrityCheckerType: 'checker1' }
+    ])).toBe('js:checker1');
+  });
+  it('serializes multiple pairs', () => {
+    expect(serializeAssetIntegrityCheckers([
+      { assetExtension: 'js', integrityCheckerType: 'checker1' },
+      { assetExtension: 'css', integrityCheckerType: 'checker2' }
+    ])).toBe('js:checker1,css:checker2');
+  });
+});
+
+describe('validateAssetIntegrityCheckers', () => {
+  it('returns true for empty string', () => {
+    expect(validateAssetIntegrityCheckers('')).toBe(true);
+  });
+  it('returns true for valid pairs', () => {
+    expect(validateAssetIntegrityCheckers('js:checker1')).toBe(true);
+    expect(validateAssetIntegrityCheckers('js:checker1,css:checker2')).toBe(true);
+  });
+  it('returns false for missing assetExtension or checkerType', () => {
+    expect(validateAssetIntegrityCheckers('js:')).toBe(false);
+    expect(validateAssetIntegrityCheckers(':checker1')).toBe(false);
+    expect(validateAssetIntegrityCheckers('js:checker1,:checker2')).toBe(false);
+    expect(validateAssetIntegrityCheckers('js:checker1,css:')).toBe(false);
+  });
+});
+
+describe('flattenRepositoryLocales', () => {
+  it('flattens hierarchical locales', () => {
+    const input = [
+      {
+        locale: 'en',
+        toBeFullyTranslated: true,
+        childLocales: [
+          { locale: 'fr', toBeFullyTranslated: false },
+          { locale: 'de', toBeFullyTranslated: false }
+        ]
+      },
+      {
+        locale: 'es',
+        toBeFullyTranslated: true,
+        childLocales: []
+      }
+    ];
+    const result = flattenRepositoryLocales(input);
+    expect(result).toEqual([
+      { locale: 'en', toBeFullyTranslated: true },
+      { locale: 'fr', toBeFullyTranslated: false, parentLocale: { locale: 'en' } },
+      { locale: 'de', toBeFullyTranslated: false, parentLocale: { locale: 'en' } },
+      { locale: 'es', toBeFullyTranslated: true }
+    ]);
+  });
+  it('handles empty input', () => {
+    expect(flattenRepositoryLocales([])).toEqual([]);
+  });
+});
+
+describe('unflattenRepositoryLocales', () => {
+  it('reconstructs hierarchy from flattened', () => {
```

Comment:
I'm curious what would happen if the parent locale's id points to something not in the data structure eg id: 42